src/ring_theory/algebra.lean

```diff
@@ -576,12 +578,64 @@ def module.restrict_scalars : module R E :=
   add_smul  := by simp [add_smul],
   zero_smul := by simp [zero_smul] }
 
+
+/--
+When `E` is a module over a ring `S`, and `S` is an algebra over `R`, then `E` inherits a
+module structure over `R`, provided as a type synonym `module.restrict_scalars R E := E`.
+-/
+def module.restrict_scalars (R : Type*) (E : Type*) : Type* := E
+instance [add_comm_group E] : add_comm_group (module.restrict_scalars R E) := by assumption
+instance : module R (module.restrict_scalars R E) :=
```

Comment:
This is still a bad instance, right?

Reply:
Yes, it was a bad instance. I've updated this now, informed by getting `Maschke` working, so that the algebra `S` is explicit too. That is, you write `module.restrict_scalars R S E` to turn an `S`-module `E` into an `R`-module, by locating an `algebra R S` instance.